gemrb/core/GameScript/GSUtils.cpp

```diff
@@ -404,6 +404,24 @@ bool HasItemCore(const Inventory *inventory, const ResRef& itemname, ieDword fla
 	return false;
 }
 
+bool RemoveItemOfStore(const ResRef& storename, const ResRef& itemname)
+{
+	Store* store = gamedata->GetStore(storename);
+	if (!store) {
+		Log(ERROR, "GameScript", "Store cannot be opened!");
+		return false;
+	}
+
+	//RemoveItem doesn't use trigger, and hopefully this will always run on bags (with no triggers)
+	unsigned int idx = store->FindItem(itemname, false);
+	if (idx == (unsigned int) -1) return false;
+	STOItem *si = store->GetItem(idx, false);
+	store->RemoveItem(si);
+	//store changed, save it
+	gamedata->SaveStore(store);
+	return true;
+}
```

Comment:
Did you plan to use this in some other function as well now? Otherwise it doesn't make much sense to be here.

Reply:
Do I should move it to Scriptable.cpp ? My first thought was doing everything on this class but I have to pass a lot of objects so I did the minimum  here to remove from store and put the rest of the logic on **Scriptable.cpp**

Reply:
First check if it can be reused, since it looks familiar. Eg. by StoreGetItemCore or something else like that. That will impact the best place to have it. A static method of Store is also an option.

Reply:
Is very similar to StoreGetItemCore the difference is the check of the amount in stock. I could convert the two of them in one with an extra parameter to be call to just remove or take in account the amount in stock.

Reply:
You'd need more, since it modifies the item by reference. So let's skip this then. Move it to Store, remove the comments (useless) and be consistent with the spacing.